sources/api/pluto/src/api.rs

```diff
@@ -1,76 +1,101 @@
-        #[snafu(display("Error calling Bottlerocket API: {}", source))]
-        ApiClient {
-            #[snafu(source(from(apiclient::Error, Box::new)))]
-            source: Box<apiclient::Error>,
-            uri: String,
-        },
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct AwsInfo {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub(crate) region: Option<String>,
+}
 
-        #[snafu(display("Unable to deserialize Bottlerocket settings: {}", source))]
-        SettingsJson { source: serde_json::Error },
-    }
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct Kubernetes {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub(crate) cluster_name: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub(crate) cluster_dns_ip: Option<modeled_types::KubernetesClusterDnsIp>,
+}
 
-    /// Gets the Bottlerocket settings from the API and deserializes them into a struct.
-    async fn get_settings() -> Result<model::Settings> {
-        let uri = constants::API_SETTINGS_URI;
-        let (_status, response_body) =
-            apiclient::raw_request(constants::API_SOCKET, uri, "GET", None)
-                .await
-                .context(ApiClientSnafu { uri })?;
+#[derive(Deserialize)]
+struct View {
+    pub aws: Option<AwsInfo>,
+    pub kubernetes: Option<Kubernetes>,
+}
 
-        serde_json::from_str(&response_body).context(SettingsJsonSnafu)
-    }
+#[derive(Deserialize)]
+struct SettingsView {
+    pub settings: View,
+}
 
-    /// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        let settings = get_settings().await?;
-        Ok(AwsK8sInfo {
-            region: settings.aws.and_then(|a| a.region).map(|s| s.into()),
-            cluster_name: settings
-                .kubernetes
-                .as_ref()
-                .and_then(|k| k.cluster_name.clone())
-                .map(|s| s.into()),
-            cluster_dns_ip: settings.kubernetes.and_then(|k| k.cluster_dns_ip),
-        })
-    }
+#[derive(Debug, Snafu)]
+pub(crate) enum Error {
+    #[snafu(display("Failed to call apiclient: {}", source))]
+    CommandFailure { source: std::io::Error },
+    #[snafu(display("APIClient execution failed: {}", reason))]
+    ExecutionFailure { reason: String },
+    #[snafu(display("Deserialization of configuration file failed: {}", source))]
+    Deserialize {
+        #[snafu(source(from(serde_json::Error, Box::new)))]
+        source: Box<serde_json::Error>,
+    },
 }
 
-/// This dummy code is compiled when the `sources` workspace is being compiled for non `aws-k8s-*`
-/// variants.
-#[cfg(not(variant_family = "aws-k8s"))]
-mod inner {
-    use super::*;
-    use snafu::Snafu;
+pub(crate) async fn client_command<I, S>(args: I) -> Result<Vec<u8>>
```

Comment:
Could you do the json parse in this function and return a `Result<serde_json::Value>`?

Reply:
I can move it here

Reply:
Yeah not sure its a good idea to change to serde_json since it is used for set as well which does not return valid json afaik